### ISSUE TITLE
IE & Edge crash with "Permission denied" on the `contents() for <frame />` test

### ISSUE THREAD

**Issue by mgol** [jquery/jquery#4441]
### Description ###

IE 11 as well as Edge (even 17-18) often crash on the [`traversing: contents() for <frame />`](https://github.com/jquery/jquery/blob/47835965bd100a3661d8299d8b769ceeb8b6ce48/test/unit/traversing.js#L883-L898) test with the "Permission denied" error. IE crashes way often than Edge. This is probably related to my Sizzle removal PR (#4395) but I wasn't able to pinpoint the exact reason so far. The error points to [line `174` in `src/selector.js`](https://github.com/jquery/jquery/blob/47835965bd100a3661d8299d8b769ceeb8b6ce48/src/selector.js#L174). An example failing run: http://swarm.jquery.org/result/3337866.

It's hard to debug as it's clearly a race condition. If I run this test in isolation, it always passes.

I think I've seen a similar failure on the `3.4-stable` branch as well but it was happening way less often.

We should figure this out before the `4.0.0` release.

### Link to test case ###

N/A, it's a core test already in the repo.

**Comment by mgol** [comment]
Unfortunately, while in the vast majority of cases the test case fails in IE, when you run that particular test in isolation it almost never fails. This makes debugging tricky.

**Comment by mgol** [comment]
Some conclusions from my investigation:

1. The error is triggered by [this line](https://github.com/jquery/jquery/blob/47835965bd100a3661d8299d8b769ceeb8b6ce48/test/unit/traversing.js#L891):
    ```js
    var contents = container.find( "#test-frame" ).contents();
    ```
1. This is not the only test failing with "Permission denied". I initially thought this one failure triggers more of them sometimes but commenting out that one test results in frequent failures in some other tests, e.g. the [`traversing: sort direction`](https://github.com/jquery/jquery/blob/47835965bd100a3661d8299d8b769ceeb8b6ce48/test/unit/traversing.js#L900-L924) test. The stack trace doesn't point to anything that could help.

**Comment by mgol** [comment]
I found the cuplrit. IE/Edge sometimes crash when comparing documents between frames... but only if one of the strict equality operators (`===` or `!==`) is used. Shallow equivalents (`==` or `!=`) work fine... PR incoming.

This means the issue might as well exist on 3.x as well, maybe it just triggers the condition less often. I'm changing the milestone to 3.4.2 & I'll submit a PR to Sizzle as well.

**Comment by mgol** [comment]
jQuery PR (for jQuery 4.0): #4471.

**Comment by mgol** [comment]
Note to self: reopen issue after landing #4471 so that a Sizzle PR also lands.

**Comment by mgol** [comment]
Reopening as we still need to land a fix in Sizzle & then update Sizzle on the `3.4-stable` branch.

**Comment by mgol** [comment]
Sizzle PR: https://github.com/jquery/sizzle/pull/459.

**Comment by mgol** [comment]
Re-opening as we still need a new Sizzle with the fix for jQuery 3.x. The Sizzle PR: https://github.com/jquery/sizzle/pull/459 has been merged.

**Comment by mgol** [comment]
Reopening since we still need to update Sizzle in jQuery 3.x.

**Comment by mgol** [comment]
Sizzle update merged in #4641. The issue is now fixed both on `master` & on `3.x-stable`.
